Dockerfile

```diff
@@ -1,17 +1,27 @@
-FROM debian:8
+FROM node:carbon-alpine as build-frontend
+WORKDIR /frontend
+COPY frontend/package.json ./
+COPY frontend/yarn.lock ./
+RUN yarn --production
+COPY frontend .
+RUN npm run build
 
-ADD backend.tar.gz /www
+FROM golang:1.9-alpine as build-backend
+WORKDIR /go/src/github.com/HackGT/SponsorshipPortal/backend
+RUN apk update && apk add git
+RUN go get -u github.com/golang/dep/cmd/dep
+RUN go get -u github.com/revel/cmd/revel
+COPY backend/Gopkg.* ./
+RUN dep ensure -vendor-only
+COPY backend .
+COPY --from=build-frontend frontend/build ./app/public
+RUN revel package github.com/HackGT/SponsorshipPortal/backend prod
+RUN pwd
+RUN ls
 
-ADD backend/textextract.rb /www
-
-ENV parsePath /www/textextract.rb
-
-RUN apt-get update && apt-get install -y \
-    ruby \
-    ruby-dev \
-    build-essential \
-    default-jre
-
-RUN gem install yomu
-
-CMD /www/run.sh
+FROM alpine:latest as run-server
+EXPOSE 9000
+WORKDIR /www
+COPY --from=build-backend /go/src/github.com/HackGT/SponsorshipPortal/backend/backend.tar.gz .
```

Comment:
If you use `ADD` instead of `COPY` it will decompress the archive for you. Does this still work with the `--from` flag?

Reply:
`ADD` does not have a `--from` flag